### PR TITLE
[main > 0.41]: Fix back-compat with old container runtime (#6498)

### PR DESCRIPTION
**Pull request by jatgarg**

### CODE REVIEW

**Review by sumedhb1995 — APPROVED** [review]

File: packages/loader/container-loader/src/containerStorageAdapter.ts [hunk 1]
```diff
@@ -29,7 +29,12 @@ export class ContainerStorageAdapter implements IDocumentStorageService {
     }
 
     public get policies(): IDocumentStorageServicePolicies | undefined {
-        return this.storageGetter().policies;
+        // back-compat 0.40 containerRuntime requests policies even in detached container if storage is present
+        // and storage is always present in >=0.41.
+        try {
+            return this.storageGetter().policies;
```
Comment (vladsud) [inline packages/loader/container-loader/src/containerStorageAdapter.ts:35]: Are you trying to do `this.storageGetter()?.policies`?
If yes, it's much better to be direct, then catch all exceptions.

Comment (jatgarg) [reply packages/loader/container-loader/src/containerStorageAdapter.ts:35]: StorageGetter throws exception if called in detached container. So we can not do "?". We need to catch that here and return undefined.

Comment (jatgarg) [reply packages/loader/container-loader/src/containerStorageAdapter.ts:35]: This is just for compat and we will remove this in couple of iterations.

**Review by vladsud — APPROVED** [review]

**Comment by anthony-murphy** [comment]
is this something we could have caught with our compat tests? Or are we missing tests?

**Comment by jatgarg** [comment]
> is this something we could have caught with our compat tests? Or are we missing tests?

I found that rehydrate container tests were running as no compat tests. Enabling them in my other PR.

**Pull request merged**